File: 2019/day1/day1.py
#!/usr/bin/env python3

def calculate_fuel(mass):
    fuel = (mass // 3) - 2
    if fuel <= 0:
        fuel = 0
    return fuel

def total_fuel(mass):
    fuel_needed = 0
    while mass > 0:
        mass = calculate_fuel(mass)
        fuel_needed += mass
    return fuel_needed


if __name__ == '__main__':
    with open('input.txt') as f:
        data = f.readlines()
        total = 0
        print(sum([total_fuel(int(mass)) for mass in data]))

File: 2019/day1/day1.go
package main

import (
	"fmt"
	"io/ioutil"
	"strconv"
	"strings"
)

func calculateFuel(mass int) int {
	fuel := (mass / 3) - 2
	if fuel <= 0 {
		fuel = 0
	}
	return fuel
}

func totalFuel(mass int) int {
	fuelNeeded := 0
	for mass > 0 {
		mass = calculateFuel(mass)
		fuelNeeded += mass
	}
	return fuelNeeded
}

func main() {
	dat, err := ioutil.ReadFile("input.txt")
	if err != nil {
		panic(err)
	}
	total := 0
	for _, mass := range strings.Split(string(dat), "\n") {
		input, err := strconv.Atoi(mass)
		if err != nil {
			continue
		}
		total += totalFuel(input)
	}
	fmt.Println(total)
}
